src/kernel/arch/x86/arch.zig

```diff
@@ -210,6 +210,54 @@ pub fn halt() void {
     asm volatile ("hlt");
 }
 
+///
+/// Return the value of control register 0.
+///
+/// Return: u32
+///     The value in CR0.
+///
+pub fn CR0() u32 {
+    return asm volatile ("mov %%cr0, %[cr0]"
+        : [cr0] "=r" (-> u32)
+    );
+}
+
+///
+/// Return the value of control register 2.
+///
+/// Return: u32
+///     The value in CR2.
+///
+pub fn CR2() u32 {
+    return asm volatile ("mov %%cr2, %[cr2]"
+        : [cr2] "=r" (-> u32)
+    );
+}
+
+///
+/// Return the value of control register 3.
+///
+/// Return: u32
+///     The value in CR3.
+///
+pub fn CR3() u32 {
+    return asm volatile ("mov %%cr3, %[cr3]"
+        : [cr3] "=r" (-> u32)
+    );
+}
+
+///
+/// Return the value of control register 4.
+///
+/// Return: u32
+///     The value in CR4.
+///
+pub fn CR4() u32 {
+    return asm volatile ("mov %%cr4, %[cr4]"
+        : [cr4] "=r" (-> u32)
+    );
+}
```

Comment:
Would you agree on moving the inline assembly from these functions into the handler itself? I think having them in here bloats the file a bit.

Reply:
But what if other files need to use the control registers

Reply:
If that happens we could abstract them out. These are so simple I think it's unnecessary to have them as functions, at the moment at least :)